.github/workflows/main.yml

```diff
@@ -1,112 +1,246 @@
-name: Code Quality Checks
+name: CI/CD
 
 on:
   push:
-  # Allows you to run this workflow manually from the Actions tab
-  workflow_dispatch:
+    branches:
+      - master
+  pull_request:
+
+env:
+  DOCKER_BUILDKIT: 1
+  COMPOSE_DOCKER_CLI_BUILD: 1
 
 jobs:
-  flake8:
-    name: flake8
+  test-e2e:
+    name: test-e2e
     runs-on: ubuntu-latest
-    container: python:2.7
+    timeout-minutes: 60
+    concurrency:
+      group: ${{ github.ref }}/test-e2e
+      cancel-in-progress: true
+    permissions:
+      id-token: write
+      contents: read
     steps:
-      - uses: actions/checkout@v2
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: setup docker buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: install nodejs v12
+        uses: actions/setup-node@v2
+        with:
+          node-version: 12
```

Comment:
Could be newer node, v12 is in end of life next april.

Reply:
Newer node won't work directly, I had it set to 16 before but the cypress project had some issues with it.

I'll leave it as a small issue for future, I'll fix it if I have time and remember.